app/models/piece.rb

```diff
@@ -5,4 +5,27 @@ class Piece < ApplicationRecord
   def icon_path
     "#{color}_#{type}.svg".downcase
   end
+
+  def create_piece
+    @piece = Piece.create(piece_params)
+  end
+
+  def move_to!(x, y)
+    opposing_piece = Piece.find_by(position_x: x, position_y: y)
+
+    if opposing_piece.present? && opposing_piece.color != color
+      opposing_piece.update_attributes(position_x: nil, position_y: nil)
+      update_attributes(position_x: x, position_y: y)
+    elsif opposing_piece.present? == false
+      update_attributes(position_x: x, position_y: y)
+    else
+      return 'invalid move'
+    end
+  end
+
+  private
+
+  def piece_params
+    params.require(:piece).permit(:position_x, :position_y, :type, :color)
+  end
```

Comment:
This is unused, can be deleted.

(generally a method like this would live in the controller)